client/src/components/Weather/index.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCloud } from '@fortawesome/free-solid-svg-icons';
+import { faCloudRain } from '@fortawesome/free-solid-svg-icons';
+import { faBolt } from '@fortawesome/free-solid-svg-icons';
+import { faSnowflake } from '@fortawesome/free-solid-svg-icons';
+import { faSun } from '@fortawesome/free-solid-svg-icons';
+import { faSmog } from '@fortawesome/free-solid-svg-icons';
+import './style.css';
+
+const API_KEY = "8e20c3b423d1a8139959af74dcb9cba2";
+const weatherIcon = {
+    Thunderstorm: <FontAwesomeIcon icon={faBolt} />,
+    Rain: <FontAwesomeIcon icon = {faCloudRain} />,
+    Snow:  <FontAwesomeIcon icon={faSnowflake} />,
+    Atmosphere: <FontAwesomeIcon icon={faSmog} />,
+    Clear: <FontAwesomeIcon icon={faSun} />,
+    Clouds: <FontAwesomeIcon icon={faCloud} />,
+}
+
+function Weather()  {
+    
+
+    const [icon,setIcon] = useState(undefined);
+    const [main,setMain] = useState(undefined);
+    const [celsius, setCelsius] = useState(undefined);
+    const [temp_max, setMaxTemp] = useState(null);
+    const [temp_min, setMinTemp] = useState(null);
+    const [description, setDescription] = useState("");
+   
+    /* the get_WeatherIcon functions set the weather icon based on the weather condition codes
+       see https://openweathermap.org/weather-conditions for more details
+       cloud drizzle icon needs pro plan, so the cloud rain icon is used instead
+    */
+    const get_WeatherIcon = (rangeId) => {
+      switch (true) {
+        case rangeId >= 200 && rangeId < 232:
+          setIcon(weatherIcon.Thunderstorm);
+          break;
+        case rangeId >= 300 && rangeId <= 321:
+          setIcon(weatherIcon.Rain);
+          break;
+        case rangeId >= 500 && rangeId <= 521:
+          setIcon(weatherIcon.Rain);
+          break;
+        case rangeId >= 600 && rangeId <= 622:
+          setIcon(weatherIcon.Snow);
+          break;
+        case rangeId >= 701 && rangeId <= 781:
+          setIcon(weatherIcon.Atmosphere);
+          break;
+        case rangeId === 800:
+          setIcon(weatherIcon.Clear);
+          break;
+        case rangeId >= 801 && rangeId <= 804:
+          setIcon(weatherIcon.Clouds);
+          break;
+        default:
+          setIcon(weatherIcon.Clouds);
+      }
+    }
+    
+  
+    const toCelsius = (temp) => {
+      let cell = Math.floor(temp - 273.15);
```

Comment:
Magic numbers should be avoided as they can confuse contributors who edit the code later on and aren't familiar with what they mean. You could try defining this number in a variable at the top of the file and then referring to that variable here, perhaps like:

```
const kelvinToCelsiusDiff = 273.15;
```
```
let cell = Math.floor(temp - kelvinToCelsiusDiff);
```